File: career_agent_backend/app/schemas/student.py
"""
Pydantic 学生相关数据模型（Schemas）

定义学生信息的请求/响应模型。
"""
from typing import Optional, Dict, Any

from pydantic import BaseModel
from datetime import datetime

from typing import Optional, Any, List


class StudentResponse(BaseModel):
    """
    学生信息响应模型。

    用于返回学生的基本信息以及解析生成的个人画像。
    """

    id: int
    """学生唯一标识 ID"""

    name: str
    """学生姓名"""

    profile_json: Optional[Dict[str, Any]] = None
    """学生画像（JSON 格式），包含技能、能力评分、教育背景等结构化信息，可选"""

    class Config:
        from_attributes = True  # 允许从 SQLAlchemy 模型实例创建 Pydantic 对象

class ResumeVersionResponse(BaseModel):
    id: int
    version: int
    created_at: datetime

    class Config:
        from_attributes = True

class ResumeVersionDetail(ResumeVersionResponse):
    resume_text: str
    profile_json: dict

class ManualBasicsUpdate(BaseModel):
    intended_city: Optional[str] = None
    gender: Optional[str] = None
    school: Optional[str] = None
    grade: Optional[str] = None

class ExperienceItem(BaseModel):
    name: str
    role: Optional[str] = None
    description: Optional[str] = None
    technologies: Optional[List[str]] = None

class ExperiencesUpdate(BaseModel):
    projects: Optional[List[ExperienceItem]] = None
    papers: Optional[List[ExperienceItem]] = None
    internships: Optional[List[ExperienceItem]] = None
    competitions: Optional[List[ExperienceItem]] = None

class StudentProfileUpdateRequest(BaseModel):
    manual_basics: Optional[ManualBasicsUpdate] = None
    experiences: Optional[ExperiencesUpdate] = None
    skills: Optional[List[str]] = None
    certificates: Optional[List[str]] = None
    education: Optional[str] = None
    major: Optional[str] = None
    age: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[str] = None
    graduation_year: Optional[str] = None
    target_job: Optional[str] = None
    self_introduction: Optional[str] = None
    language: Optional[str] = None
    work_experience: Optional[str] = None
    internships: Optional[List[str]] = None
    innovation_score: Optional[float] = None
    innovation_reason: Optional[str] = None
    learning_score: Optional[float] = None
    learning_reason: Optional[str] = None
    stress_score: Optional[float] = None
    stress_reason: Optional[str] = None
    communication_score: Optional[float] = None
    communication_reason: Optional[str] = None
    overall_score: Optional[float] = None
    overall_reason: Optional[str] = None
    confidence_score: Optional[float] = None
    confidence_reason: Optional[str] = None
    interests: Optional[List[str]] = None
    strengths: Optional[List[str]] = None

class FreeTextParseRequest(BaseModel):
    text: str
    type: str  # "project" | "paper" | "internship" | "competition" | "basic"

File: career_agent_frontend/src/api/student.ts
// src/api/student.ts
import request from '../utils/request'

// 类型定义（如果已在 types/api.ts 中定义，可导入）
export interface StudentResponse {
  id: number
  name: string
  profile_json: Record<string, any> | null
}

export interface ResumeVersionResponse {
  id: number
  version: number
  created_at: string
}

export interface ResumeVersionDetail extends ResumeVersionResponse {
  resume_text: string
  profile_json: Record<string, any>
}

export interface StudentProfileUpdateRequest {
  manual_basics?: {
    intended_city?: string | null;
    gender?: string | null;
    school?: string | null;
    grade?: string | null;
  } | null;
  experiences?: {
    projects?: Array<{ name: string; role?: string; description?: string; technologies?: string[] }> | null;
    papers?: Array<{ name: string; role?: string; description?: string; technologies?: string[] }> | null;
    internships?: Array<{ name: string; role?: string; description?: string; technologies?: string[] }> | null;
    competitions?: Array<{ name: string; role?: string; description?: string; technologies?: string[] }> | null;
  } | null;
  skills?: string[] | null;
  certificates?: string[] | null;
  education?: string | null;
  major?: string | null;
  // 以下为新增字段，与后端 StudentProfileUpdateRequest 保持一致
  age?: string | null;
  phone?: string | null;
  email?: string | null;
  graduation_year?: string | null;
  target_job?: string | null;
  self_introduction?: string | null;
  interests?: string[] | null;
  strengths?: string[] | null;
}

export interface FreeTextParseRequest {
  text: string
  type: string
}

// 上传简历
export const uploadResume = (file: File) => {
  const formData = new FormData()
  formData.append('file', file)
  return request.post<StudentResponse>('/api/v1/students/upload', formData, {
    headers: { 'Content-Type': 'multipart/form-data' }
  })
}

// 获取学生信息
export const getStudent = (studentId: number) => {
  return request.get<StudentResponse>(`/api/v1/students/${studentId}`)
}

// 更新学生画像
export const updateStudentProfile = (studentId: number, data: StudentProfileUpdateRequest) => {
  return request.put<StudentResponse>(`/api/v1/students/${studentId}/profile`, data)
}

// 确认生成完整画像
export const finalizeStudent = (studentId: number) => {
  return request.post<StudentResponse>(`/api/v1/students/${studentId}/finalize`)
}

// 获取简历版本列表
export const getResumeVersions = (studentId: number) => {
  return request.get<ResumeVersionResponse[]>(`/api/v1/students/${studentId}/versions`)
}

// 获取指定版本详情
export const getResumeVersion = (studentId: number, versionId: number) => {
  return request.get<ResumeVersionDetail>(`/api/v1/students/${studentId}/versions/${versionId}`)
}

// 解析自由文本
export const parseText = (data: FreeTextParseRequest) => {
  return request.post<any>('/api/v1/students/parse-text', data)
}